### PR TITLE
Feature: hooks

### PR DESCRIPTION
**Pull request by allanhortle**

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+packages/enty-prototype/* linguist-generated
```
Comment (thepont) [inline .gitattributes]: Is using linguist-generated for executable code a good idea?

Comment (allanhortle) [reply .gitattributes]: It's not published. It's just a scratch pad.

Comment (allanhortle) [reply .gitattributes]: I plan to remove it soon.

File: .gitignore [hunk 2]
```diff
@@ -7,6 +7,7 @@
 .cache
 *.orig
 *.log
+*.un~
```
Comment (dxinteractive) [inline .gitignore]: What is this?

File: packages/react-enty-hooks/src/RequestHook.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        });
+
+
+        return useMemo(() => new Message({
+            resultKey,
+            requestState,
+            response,
+            requestError: state._error[resultKey],
+            onRequest
+        }), [requestState, response]);
```
Comment (dxinteractive) [inline packages/react-enty-hooks/src/RequestHook.js]: Should `resultKey` be in this array / are there any cases where `resultKey` might change that doesn't also guarantee a change in `requestState` or `response`?

File: packages/react-enty-hooks/src/__test__/EntityApi-test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                }
+            }
+        });
+    });
+
+});
+
+describe('Provider', () => {
+
+    it('will transparently stack providers', () => {
```
Comment (dxinteractive) [inline packages/react-enty-hooks/src/__test__/EntityApi-test.js]: Missing test assertions?

File: packages/react-enty-hooks/src/__test__/RequestHoc-test.js [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+        expect(typeof foo.requestHoc({name: 'foo'})).toBe('function');
+    });
+
+    describe('config.name and message', () => {
+
+        test('will throw an error if config.name is not supplied', () => {
+            expect(() => foo.requestHoc({})).toThrow('requestHoc must be given a name');
+        });
+
+        it('will memoise the creation of the message based on requestState and response', () => {
```
Comment (dxinteractive) [inline packages/react-enty-hooks/src/__test__/RequestHoc-test.js]: Missing test?

**Comment by allanhortle** [comment]
Test comment.

File: .gitignore (hunk 2 above)
Comment (allanhortle) [reply .gitignore]: It's a vim undo tree file. (Commenting late for hubgit testing)

Comment (allanhortle) [reply .gitignore]: ...Extra reply for testing

Comment (allanhortle) [reply .gitignore]: Checking if unresolved stuff works.

**Pull request merged**